pages/common/sg.md

```diff
@@ -0,0 +1,20 @@
+# sg
+
+> Ast-grep is a tool for code structural search, lint, and rewriting.
+> More information: <https://ast-grep.github.io/guide/introduction.html>.
+
+- Scan possible queries using interactive mode:
+
+`sg scan --interactive`
+
+- Rewrite code in the current directory using patterns:
+
+`sg run --pattern {{'foo'}} --rewrite {{'bar'}} --lang {{python}}`
+
+- Visualize possible changes without applying them:
+
+`sg -p 'useState<number>($A)' -r 'useState($A)' -l ts`
```

Comment:
[long options](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax) and placeholders
```suggestion
- Visualize possible changes without applying them:

`sg run --pattern '{{useState<number>($A)}}' --rewrite '{{useState($A)}}' --lang {{typescript}}`
```